frontend/src/components/graphs/SuiteInstability.js

```diff
@@ -0,0 +1,276 @@
+                states += 1;
+            }
+
+            if (testRun.status === 'PASS') {
+                passes++;
+            }
+        });
+
+        return passes / states / test['builds'].length;
+    };
+
+    const generateBarData = historyDataState => {
+        const suites = [];
+
+        historyDataState['history'].forEach(suite => {
+            const failingTests = [];
+
+            suite['test_cases'].forEach(testCase => {
+                const isFailingTest = testCase['builds'].some(testRun => {
+                    return testRun['status'] === 'FAIL';
+                });
+                if (isFailingTest) {
+                    const stability = calculateTestStability(testCase);
+                    failingTests.push({ name: testCase.name, stability });
+                }
+            });
+
+            if (failingTests.length) {
+                suites.push({
+                    id: suite['suite_id'],
+                    name: suite['name'],
+                    failingTests,
+                });
+            }
+        });
+
+        const failingSuites = suites.map(suite => {
+            const stability =
+                suite.failingTests.reduce(
+                    (acc, test) => acc + test.stability,
+                    0
+                ) / suite.failingTests.length;
+
+            return {
+                name: suite.name,
+                numberOfFailingTests: suite.failingTests.length,
+                stability,
+                id: suite.id,
+            };
+        });
+
+        return { failingSuites };
+    };
+
+    if (!historyDataState || loadingState) {
+        return <Loading />;
+    }
+
+    const selectSuite = id => {
+        const suite = historyDataState['history'].find(
+            suite => suite['suite_id'] === id
+        );
+
+        if (suite) {
+            setSelectedSuite(suite);
+        }
+    };
+
+    const deselectSuite = () => setSelectedSuite(null);
+
+    const handleBarChartClick = (_name, values) => {
+        if (values.id) {
+            selectSuite(values.id[0]);
+        } else {
+            deselectSuite();
+        }
+    };
+
+    const signalListeners = { select: handleBarChartClick };
+
+    const barData = generateBarData(historyDataState);
+
+    const buildsInTotal = Math.min(
+        historyDataState['max_build_num'],
+        numberOfBuilds
+    );
+
+    const generateStatusRow = testCase => {
+        let statuses = [];
+
+        for (let i = 0; i < buildsInTotal; i++) {
+            const testStatus = testCase['builds'].find(build => {
+                return build['build_number'] === i + 1;
+            });
+
+            if (testStatus) {
+                if (testStatus['status'] === 'PASS') {
+                    statuses.push('x');
+                } else if (testStatus['status'] === 'FAIL') {
+                    statuses.push('o');
+                }
+            } else {
+                // Impute
```

Comment:
?

Reply:
This is actually JP:s comment, so don't know the precise reason